File: src/luogu/p1093/Main.cpp
// https://www.luogu.com.cn/problem/P1093
#include <algorithm>
#include <iostream>
#include <string>
using namespace std;

typedef struct student {
  int num;
  int chi, mat, eng;
  int sum;

  bool operator<(const student& b) {
    if (sum > b.sum) {
      return true;
    } else if (sum < b.sum) {
      return false;
    } else {
      if (chi > b.chi) {
        return true;
      } else if (chi < b.chi) {
        return false;
      } else {
        if (num < b.num) {
          return true;
        } else {
          return false;
        }
      }
    }
  }
} student;

int main() {
  int n;
  cin >> n;
  student s[n];
  for (int i = 0; i < n; i++) {
    s[i].num = i + 1;
    cin >> s[i].chi >> s[i].mat >> s[i].eng;
    s[i].sum = s[i].chi + s[i].mat + s[i].eng;
  }
  sort(s, s + n);
  for (int i = 0; i < 5; i++) {
    cout << s[i].num << ' ' << s[i].sum << endl;
  }
  return 0;
}

File: src/luogu/p1093/Main.java
// https://www.luogu.com.cn/problem/P1093
package luogu.p1093;

import java.util.Arrays;
import java.util.Scanner;

public class Main {
  static class Student {
    int id;
    int chi, mat, eng;
    int sum;
  }

  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    int n = sc.nextInt();
    Student[] s = new Student[n];
    for (int i = 0; i < n; i++) {
      s[i] = new Student();
      s[i].id = i + 1;
      s[i].chi = sc.nextInt();
      s[i].mat = sc.nextInt();
      s[i].eng = sc.nextInt();
      s[i].sum = s[i].chi + s[i].mat + s[i].eng;
    }
    Arrays.sort(s, (a, b) -> {
      if (a.sum > b.sum) {
        return -1;
      } else if (a.sum < b.sum) {
        return 1;
      } else {
        if (a.chi > b.chi) {
          return -1;
        } else if (a.chi < b.chi) {
          return 1;
        } else {
          if (a.id < b.id) {
            return -1;
          } else {
            return 1;
          }
        }
      }
    });
    for (int i = 0; i < 5; i++) {
      System.out.println(s[i].id + " " + s[i].sum);
    }
  }
}
